### PR TITLE
DCMAW-9802: Refactor session creation

### PR DESCRIPTION
**Pull request by J-son1**
​DCMAW-9802

### What changed

- Refactors creating a session to reduce the length of the /credential Lambda by moving logic to the Session Service.

## Checklists

- [x] There is a ticket raised for this PR that is present in the branch name
- [x] No PII data logged. [See guidance here](https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3502407722/PII+Logging+Considerations)
- [ ] Demo to a BA, TA, and the team.
- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: backend-api/src/functions/asyncCredential/asyncCredentialHandler.ts [hunk 1]
```diff
@@ -221,9 +207,9 @@ export async function lambdaHandler(
 
   const writeEventResult = await eventService.writeGenericEvent({
     eventName: "DCMAW_ASYNC_CRI_START",
-    sub,
+    sub: requestBody.sub,
     sessionId,
-    govukSigninJourneyId: govuk_signin_journey_id,
+    govukSigninJourneyId: requestBody.govuk_signin_journey_id,
```
Comment (J-son1) [inline backend-api/src/functions/asyncCredential/asyncCredentialHandler.ts:212]: Opted to use these values inline as opposed to deconstructing above to as there are only two values needed - avoids adding an extra line

File: backend-api/src/functions/asyncCredential/sessionService/sessionService.ts [hunk 2]
```diff
@@ -72,31 +73,34 @@ export class SessionService implements IGetActiveSession, ICreateSession {
...
-        sessionState: { S: sessionConfig.sessionState },
-        issuedOn: { S: sessionConfig.issuedOn },
+        sessionId: { S: sessionId },
+        state: { S: state },
+        sub: { S: sub },
+        clientId: { S: client_id },
+        govukSigninJourneyId: { S: govuk_signin_journey_id },
+        issuer: { S: issuer },
+        sessionState: { S: "ASYNC_AUTH_SESSION_CREATED" },
+        issuedOn: { S: Date.now().toString() },
```
Comment (J-son1) [inline backend-api/src/functions/asyncCredential/sessionService/sessionService.ts:93]: Should we have these values as default parameters on the function?

File: backend-api/src/functions/asyncCredential/sessionService/sessionService.test.ts [hunk 3]
```diff
@@ -109,17 +109,17 @@ describe("Session Service", () => {
...
-          sub: "mockSub",
-          clientId: "mockClientId",
-          govukSigninJourneyId: "mockJourneyId",
-          redirectUri: "https://mockRedirectUri.com",
-          issuer: "mockIssuer",
-          sessionState: "mockSessionState",
-          issuedOn: "mockIssuedOn",
-        });
+        const result = await service.createSession(
+          "137d5a4b-3046-456d-986a-147e0469cf62",
```
Comment (sandy-may) [inline backend-api/src/functions/asyncCredential/sessionService/sessionService.test.ts:113]: Similar here - can we calculate this outside of the lambda?

File: backend-api/src/functions/asyncCredential/sessionService/sessionService.ts [hunk 4]
```diff
@@ -72,31 +73,39 @@ export class SessionService implements IGetActiveSession, ICreateSession {
...
+    const {
+      state,
+      sub,
+      client_id,
+      govuk_signin_journey_id,
+      redirect_uri,
+      issuer,
+    } = config;
+
+    const putItemCommandConfig: IPutAuthSessionConfig = {
```
Comment (sandy-may) [inline backend-api/src/functions/asyncCredential/sessionService/sessionService.ts:88]: naming here isn't consistent with the interface

**Review by sandy-may — DISMISSED** [review]

File: backend-api/src/functions/asyncCredential/sessionService/sessionService.test.ts (hunk 3 above)
Comment (J-son1) [reply backend-api/src/functions/asyncCredential/sessionService/sessionService.test.ts]: Updated

File: backend-api/src/functions/asyncCredential/sessionService/sessionService.ts (hunk 4 above)
Comment (J-son1) [reply backend-api/src/functions/asyncCredential/sessionService/sessionService.ts]: updated

File: backend-api/src/functions/asyncCredential/sessionService/sessionService.ts [hunk 5]
```diff
@@ -85,7 +85,7 @@ export class SessionService implements IGetActiveSession, ICreateSession {
       issuer,
     } = config;
 
-    const putItemCommandConfig: IPutAuthSessionConfig = {
+    const putAuthSessionConfig: IPutAuthSessionConfig = {
```
Comment (sandy-may) [inline backend-api/src/functions/asyncCredential/sessionService/sessionService.ts:88]: nit: keen to take away the use of 'Auth' if possible please :)

**Review by sandy-may — APPROVED** [review]

**Pull request merged**